mattermost/mattermost-mobile
# Detox/E2E: Evaluate and create POC for BitBar/Detox Mobile E2E

#### Summary
- This PR is Phase 1 of the POC (See ticket for a general overview of approach)
- The goal for this phase is to be able to submit test files to BitBar and have a successful BitBar run
- Prerequisite: Load any emulator from AVD manager with API Level 29 or above OR better yet attach an actual Android Device to your computer and make sure it is read by `adb devices`
- In order to test:
1. Create a free BitBar account at https://bitbar.com/signup/
2. Build release APK from `mattermost-mobile`
```
cd detox
npm i
npm run e2e:android-build-release
```
3. Build test zip files, run locally, and verify test ran successfully (right now, only one test is being invoked by the script which is `e2e/test/smoke_test/login_email.e2e.js`)
```
cd ../bitbar
./zip-test-files-android.sh
./run-tests-android.sh
```
4. Upload to app APK, androidTest APK, and test zip file to BitBar, then create and run automated test
![Screen Shot 2022-01-20 at 9 35 03 AM](https://user-images.githubusercontent.com/487991/150391824-c2047eb0-dcf0-4614-9e66-2c79c87bbc86.png)
5. Check logs by clicking `Go to results -->` button
![Screen Shot 2022-01-20 at 9 37 09 AM](https://user-images.githubusercontent.com/487991/150392060-681bcdac-139a-41b4-9231-45dc01de171b.png)

**Current Issue to be resolved:**
If you look at the logs generated (scroll down towards near the bottom), you will see that the error indicates Detox is not able to connect to the app. If you can help investigate why it is so, that would be awesome.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40966

#### Release Note
```release-note
NONE
```
